src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,516 @@
+<?php
+/**
+ * Autoloader for WordPress classes.
+ *
+ * Include this file if you'd like to avoid having to create your own autoloader.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Autoloader.
+ */
+final class WP_Autoload {
```

Comment:
I think this will need something to load all the classes regardless.

In the past there have been issues with the upgrade process in which classes haven't been preloaded and fatal errors occurring due to order of operations issues in which some autoloaded classes are the old version (ie, the version of WP being upgraded from) and some autoloaded classes are the new version (ie, the version of WP being upgraded to).

@costdev has domain knowledge of this.

Reply:
Thanks for the ping @peterwilsoncc!

Since `wp-admin/includes/update-core.php` is loaded from the "new" version, if the "new" version has changed filename (case changes will also catch you here on case-sensitive/insensitive filesystems), class name, namespace for a class that's used during the upgrade process, that class will need to be preloaded before the upgrade process copies the "new" version's files into place (circa Line 1332 in `wp-admin/includes/update-core.php` at the moment).

Reply:
Hmmm just to be sure I understand what you're proposing:
* Add a `preload_all()` method which will simply `require_once` all files in the classMap
* Call `WP_Autoload::preload_all()` right before the lines on https://github.com/WordPress/wordpress-develop/blob/a9c985cde45a37fe9b329210a8eac3c0fb89c79b/src/wp-admin/includes/update-core.php#L1331-L1332

Is that correct?

Reply:
Rather than preloading all of the classes, I'd suggest that we compile a list of required classes for the upgrader and preload those.

You can either place the resulting function just above those lines, or just below the call to `_preload_old_requests_classes_and_interfaces( $to );` near the top of `update_core()`.

Reply:
Thank you @costdev!

In https://github.com/WordPress/wordpress-develop/pull/3470/commits/ba7f83039ad6a1cf7ba2684c28f64dd2e460fe5d I added a `_preload_core_classes_and_interfaces()` function, with a list of classes that should be _preloaded_.
I took a first stab at identifying those classes, but I'm sure I missed many.
Do you know if perhaps we have somewhere a list of classes involved in Core upgrades already? The safe thing would be to just load everything, but that would be inefficient and inelegant

Reply:
> Do you know if perhaps we have somewhere a list of classes involved in Core upgrades already?

I don't think so, or if we do, I don't remember where at the moment. 😅